### PR TITLE
Set up Sonatype OSS deployment

### PR DESCRIPTION
**Pull request by dflemstr**

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -184,7 +155,104 @@
...
+      <activation>
+        <property>
+          <name>performRelease</name>
+          <value>true</value>
+        </property>
+      </activation>
+      <build>
+        <plugins>
+          <plugin>
+            <artifactId>maven-source-plugin</artifactId>
```
Comment (dflemstr) [inline pom.xml]: This uses a newer version of the plugins, and as it says in the super pom: the default release profile is deprecated.

File: .travis.yml [hunk 2]
```diff
@@ -0,0 +1,15 @@
+language: java
+jdk: oraclejdk8
+
+sudo: false
+install: true # This runs the "true" command in the install phase; thus skipping install
+script:
+  - |
```
Comment (udoprog) [inline .travis.yml]: Write as script to make less cluttered?

File: pom.xml [hunk 3]
```diff
@@ -184,7 +155,104 @@
...
+          <tagNameFormat>v@{project.version}</tagNameFormat>
+          <allowTimestampedSnapshots>true</allowTimestampedSnapshots>
+          <autoVersionSubmodules>true</autoVersionSubmodules>
+        </configuration>
+        <dependencies>
+          <dependency>
+            <groupId>org.apache.maven.scm</groupId>
+            <artifactId>maven-scm-provider-gitexe</artifactId>
+            <version>1.9</version>
+          </dependency>
```
Comment (udoprog) [inline pom.xml]: Why is this needed?
